virttest/graphical_console.py

```diff
@@ -96,6 +147,104 @@ def _hold_time(self, hold_time):
         """
         time.sleep(hold_time)
 
+    def btn_press(self, btnstroke, interval=0):
+        """
+        Press the button.
+
+        :param btnstroke: button, like left, right, middle.
+        :param interval: the interval between two buttons, default interval=0.
+        """
+
+        for btn in self._get_seq(btnstroke, interval):
+            self._btn_event(self.BTN_DOWN, btn)
+
+    def btn_release(self, btnstroke, interval=0):
+        """
+        Release the button.
+
+        :param btnstroke: button, like left, right, middle.
+        :param interval: the interval between two buttons, default interval=0.
+        """
+
+        for btn in self._get_seq(btnstroke, interval):
+            self._btn_event(self.BTN_UP, btn)
+
+    def btn_click(self, btnstroke, hold_time=0, interval=0):
+        """
+        Click the button.
+
+        :param btnstroke: Mouse button, like left, right, middle.
+        :param hold_time: the time that keep one btn pressed, default hole_time=0.
+        :param interval: the interval between two btns, default interval=0.
+        """
+
+        self.btn_press(btnstroke, interval)
+        self._hold_time(hold_time)
+        btnstroke = self._reverse(btnstroke)
+        self.btn_release(btnstroke, interval)
+
+    def scroll_backward(self, count=1, interval=0):
+        """
+        Scroll the wheel to backward.
+
+        :param count: scroll count, default count=1.
+        :param interval: the interval between two scroll actions, default interval=0.
+        """
+
+        count -= 1
+        for num in range(count):
+            self._scroll_event(self.SCROLL_BACKWARD)
+            if interval:
+                time.sleep(interval)
+        self._scroll_event(self.SCROLL_BACKWARD)
+
+    def scroll_forward(self, count=1, interval=0):
+        """
+        Scroll the wheel to forward.
+
+        :param count: scroll count, default count=1.
+        :param interval: the interval between two scroll actions, default interval=0.
+        """
+
+        count -= 1
+        for num in range(count):
+            self._scroll_event(self.SCROLL_FORWARD)
+            if interval:
+                time.sleep(interval)
+        self._scroll_event(self.SCROLL_FORWARD)
+
+    def pointer_move(self, pos, motion=None, _abs=True):
```

Comment:
Please use the full spelling `absolute` instead of `_abs`.

Reply:
ok, will update~